pkg/merkle/tree.go

```diff
@@ -303,6 +293,7 @@ func roundUpToPowerOf2(n int) int {
 	n |= n >> 4
 	n |= n >> 8
 	n |= n >> 16
+	n |= n >> 32
```

Comment:
why is this change here?

Reply:
Just allows the `roundUpToPowerOf2` to be accurate for even larger numbers.

Reply:
`int` is variable sized. You are right that in most cases it will be 64 bit and the shift allows for larger powers of two. On a 32bit system this is undefined behavior. You would need `bits.OnesCount64` as well. I have to go look over your entire changeset, but in the existing code we made sure that the tree never grows beyond 32 bits. There were checks in all relevant places to ensure that the code is correct on both architectures.

Reply:
It's there a way to ensure correctness in the event that this runs on either 32-bit or 64-bit architectures? If we're targeting 32-bit, then I can make the relevant changes on the solidity and js side as well.